packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -24,56 +32,146 @@ const useStyles = makeStyles()((theme) => {
+                <img
+                    src={theme.palette.mode === 'dark' ? SettingIconDarkModeUrl : SettingIconLightModeUrl}
+                    className={classes.settingIcon}
+                    onClick={() => setOpenSettings(true)}
+                />
+            </div>
 
-                            return acc.concat(
-                                cur.ApplicationEntries.map((x) => {
-                                    return {
-                                        entry: x,
-                                        enabled: isSNSEnabled && (account ? isWeb3Enabled : !isWalletConnectedRequired),
-                                        pluginId: cur.ID,
-                                    }
-                                }) ?? [],
-                            )
-                        },
-                        [],
-                    )
-                    .sort((a, b) => a.entry.defaultSortingPriority - b.entry.defaultSortingPriority)
-                    .map((X, i) => {
+            {loading ? (
+                <div className={classes.loadingWrapper}>
+                    <CircularProgress size={24} color="primary" sx={{ marginRight: 1 }} />
+                </div>
+            ) : listedAppList.length > 0 ? (
+                <section className={classes.applicationWrapper}>
+                    {listedAppList.map((application) => {
                         return (
-                            <Fragment key={i + X.pluginId}>
-                                <X.entry.RenderEntryComponent disabled={!X.enabled} />
+                            <Fragment key={application.pluginId}>
+                                <RenderEntryComponentWrapper application={application} />
                             </Fragment>
                         )
                     })}
-            </section>
+                </section>
+            ) : (
+                <div className={classes.placeholderWrapper}>
+                    <Typography className={classes.placeholder}>
+                        {t('application_settings_tab_plug_app-unlisted-placeholder')}
+                    </Typography>
+                </div>
+            )}
+            {openSettings ? (
+                <ApplicationSettingDialog
+                    open={openSettings}
+                    onClose={() => {
+                        setOpenSettings(false)
+                        retry()
+                    }}
+                />
+            ) : null}
         </>
     )
 }
+
+interface RenderEntryComponentWrapperProps {
+    application: Application
+}
+
+function RenderEntryComponentWrapper({ application }: RenderEntryComponentWrapperProps) {
+    const RenderEntryComponent = application.entry.RenderEntryComponent!
+    return (
+        <RenderEntryComponent
+            disabled={!application.enabled}
+            icon={application.entry.icon}
+            title={application.entry.name}
```

Comment:
You should use `<PluginI18NFieldRender field={label} pluginID={id} />` or `usePluginI18NField`